src/main/java/school/lemon/changerequest/java/io/hw1/IFileManagerImp.java

```diff
@@ -0,0 +1,625 @@
+    public boolean delete(String name, boolean force) throws FileNotFoundException {
+        try
+        {
+            File file = new File(name);
+            if(file.exists())
+            {
+                if(file.isDirectory()&&file.length()>0&&force)
+                {
+                    file.delete();
+                    return true;
+                }
+                else if(file.isDirectory()&&file.length()>0&&force) return false;
+            }
+            else
+            {
+                throw new FileNotFoundException();
+            }
+        }
+        catch (FileNotFoundException fnfe)
+        {
+            System.out.println(fnfe.getMessage());
+        }
+        return false;
+    }
+
+
+
+
+    /**
+     * Copy file from source to dest using copyMode.
+     *
+     * @param source   source file to copy
+     * @param dest     destination file to copy
+     * @param copyMode copyMode
+     * @throws FileNotFoundException     if source file doesn't exist
+     * @throws IllegalArgumentException  if source file is directory
+     * @throws FileAlreadyExistException if dest file already exist
+     * @throws FileCopyException         if any exception happened during copying
+     * @see CopyMode
+     */
+    @Override
+    public void copy(String source, String dest, CopyMode copyMode) throws FileNotFoundException, IllegalArgumentException, FileAlreadyExistException, FileCopyException {
+        switch (copyMode) {
+
+
+            case SIMPLE:  try {
+                File sourceFile = new File(source);
+                File destFile = new File(dest);
+                if (!(sourceFile.exists())) throw new FileNotFoundException();
+                if (sourceFile.isDirectory()) throw new IllegalArgumentException();
+                if (destFile.exists()) throw new FileAlreadyExistException(destFile);
+                else destFile.createNewFile();
+                FileInputStream fin = new FileInputStream(source);
+                FileOutputStream fout = new FileOutputStream(dest);
+                int i;
+                do {
+                    i = fin.read();
+                    if (i != -1)
+                        fout.write(i);
+                }
+                while (i != -1);
+                fin.close();
+                fout.close();
+                System.out.println("File copied successfully! Using Simple Mode");
+            } catch (FileNotFoundException | IllegalArgumentException | FileAlreadyExistException n) {
+                System.out.println(n.getMessage());
+            } catch (IOException n) {
+            }
+            break;
+            case BUFFERED:
+
+                    copy(source, dest);
+
+                System.out.println("File copied successfully! Using Simple Buffered Mode");
+break;
+        }
+    }
+
+    @Override
+    public void copy(String source, String dest) throws FileNotFoundException, IllegalArgumentException, FileAlreadyExistException, FileCopyException {
```

Comment:
Move this implementation into `copy(String source, String dest, CopyMode copyMode)` and just call it from this method with correct parameters.
This also will reduce code duplication for file existence checks, etc.